pkg/deploy/kustomize.go

```diff
@@ -0,0 +1,115 @@
+package deploy
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const yamlBufferSize = 4096
+
+// ApplyKustomizeManifests renders manifests via Kustomize and
+// reconciles each resource in the cluster using server-side apply.
+// Accepts a customizable file system and Kustomizer to support in-memory testing.
+func ApplyKustomizeManifests(
+	ctx context.Context,
+	cli client.Client,
+	scheme *runtime.Scheme,
+	fs filesys.FileSystem,
+	k *krusty.Kustomizer,
```

Comment:
Why do we need to pass kustomizer argument? Should we be initializing in the function similar to what's being done in the tests?

```
	k := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
```